src/actions/recovery.ts

```diff
@@ -66,10 +65,11 @@ export const onRestoreAccess = (mnemonicInput: string[]): ThunkAction => async (
   if (recovered) {
     const resetServiceValuesInKeychain = useResetKeychainValues(PIN_SERVICE)
     await resetServiceValuesInKeychain()
+    dispatch(navigationActions.navigateBackHome())
   }
 
-  await dispatch(checkLocalDeviceAuthSet)
-  return dispatch(navigationActions.navigatorResetHome())
+  // we re-lock the app, which will trigger the create pin screen
+  return dispatch(genericActions.lockApp())
```

Comment:
When recovery is successful this will go to Home => Lock => Register PIN, it all changes on the screen within a second, and give quite a visual overload. 
A workaround will be to pass to Lock props that user will be redirected to after app unlock conditionally